src/element.ts

```diff
@@ -9,22 +9,42 @@
 import { Vex } from './vex';
 import { Registry } from './registry';
 import { Flow } from './tables';
+import { BoundingBox } from './boundingbox';
+import { Font } from './smufl';
+import { RenderContext, Style, ElementAttributes } from './types/common';
 
-export class Element {
-  static newID() {
-    return 'auto' + Element.ID++;
+export abstract class Element {
+  protected static ID: number = 1000;
+
+  protected context?: RenderContext;
+
```

Comment:
do we need all these line breaks among properties?

Reply:
My preference is no line breaks. (Breaks between groups of related members is okay.)